### PR TITLE
The Factory generation task does not meet the needs of the LSD project, as it treats null strings and empty strings the same way.

### PR DESCRIPTION
**Pull request by yahiaoui97**
The Factory generation task does not meet the needs of the LSD project, as it treats null strings and empty strings the same way.

### CODE REVIEW

File: galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/generator/FactoryGenerator.kt [hunk 1]
```diff
@@ -184,7 +183,7 @@ class FactoryGenerator {
       factory.fileExtension
     )
     writer = IOUtil.getFactoryWriter(output)
-    printer.print(factory, writer, options.getAbstract!!)
+    printer.print(factory, writer, options.getAbstract!!, options.lsdFactory ?: false)
```
Comment (RiadhCherni) [inline galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/generator/FactoryGenerator.kt:186]: Merci de changer 
"options.lsdFactory ?: false" à options.lsdFactory!!

File: galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/options/FactoryGeneratorOptions.kt
Comment (RiadhCherni) [inline galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/options/FactoryGeneratorOptions.kt:1]: Merci de changer l'initialisation du variable total à :
val total = arrayOfNulls<String>(parent.size + 7)

File: galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/options/FactoryGeneratorOptions.kt [hunk 2]
```diff
@@ -71,12 +76,13 @@ class FactoryGeneratorOptions(var name: String? = null,
       total[parent.size + 3] = "  --directory, -d<String>: Target directory for generated Factory files."
       total[parent.size + 4] = "  --classpath, -c<String>: Classpath specifying classes to include during compilation. pathA;pathB;pathC — Class search path of directories and JAR files."
       total[parent.size + 5] = "  --getAbstract, -a<boolean>: Generate methods for abstract types"
+      total[parent.size + 6] = "  --lsdFactory, -a<boolean>: Generate methods for LSD Project"
```
Comment (RiadhCherni) [inline galite-util/src/main/kotlin/org/kopi/galite/util/xsdToFactory/options/FactoryGeneratorOptions.kt:79]: Merci de changer "-a" à "-l"

**Review by RiadhCherni — CHANGES_REQUESTED** [review]

**Review by RiadhCherni — APPROVED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
 org.gradle.jvmargs=-Xmx4g
 #
 group=org.kopi
-version=1.5.4
+version=1.5.4-02DD
```
Comment (mgrati) [inline gradle.properties:4]: In the final branch to integrate, gradle.properties should stay inchanged.

**Review by mgrati — CHANGES_REQUESTED** [review]
We should rename the option lsdFactory. In fact, Galite is more general and shouldn't recognize LSD. We should give the option a significant name that indicate what is does : We can name it for example keepEmptyStrings.

**Review by mgrati — APPROVED** [review]
Approved changes.

**Pull request merged**